<a href="https://colab.research.google.com/github/sdasmaha/Ml_Engineers_445/blob/master/Diabetes_Notebook_Exercise1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

 Notebook Excercise #1 Part2 
 SRINJOY DASMAHAPATRA

 IDE => Google Colab


#Importing Libraries

In [38]:
import numpy as np
import pandas as pd

#Mounting the Google Drive


In [39]:
from google.colab import drive
drive.mount("/content/gdrive")

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).



#Loading the Dataset


In [40]:
diabetes_df = pd.read_csv('/content/gdrive/My Drive/Colab Notebooks/BAJWA-ML/diabetes_dataset.csv')

# *2.1*

**A**

In [41]:
print(diabetes_df.shape) 

(768, 9)


In [ ]:
print(diabetes_df.axes)

[RangeIndex(start=0, stop=768, step=1), Index(['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Insulin',
       'BMI', 'DiabetesPedigreeFunction', 'Age', 'Outcome'],
      dtype='object')]


In [ ]:
print(diabetes_df.dtypes)

Pregnancies                   int64
Glucose                       int64
BloodPressure                 int64
SkinThickness                 int64
Insulin                       int64
BMI                         float64
DiabetesPedigreeFunction    float64
Age                           int64
Outcome                       int64
dtype: object


**B**

In [ ]:
diabetes_df.head(10)

Pregnancies  Glucose  BloodPressure  ...  DiabetesPedigreeFunction  Age  Outcome
0            6      148             72  ...                     0.627   50        1
1            1       85             66  ...                     0.351   31        0
2            8      183             64  ...                     0.672   32        1
3            1       89             66  ...                     0.167   21        0
4            0      137             40  ...                     2.288   33        1
5            5      116             74  ...                     0.201   30        0
6            3       78             50  ...                     0.248   26        1
7           10      115              0  ...                     0.134   29        0
8            2      197             70  ...                     0.158   53        1
9            8      125             96  ...                     0.232   54        1

[10 rows x 9 columns]

**C**    Based on the knowledge of the dataset and its stated usage, we are dealing with supervised problem. 

Since in a supervised learning model, the algorithm learns on a labeled dataset, providing an answer key that the algorithm can use to evaluate its accuaracy on training data. 

On the contrary the unsupervised model works on unlabeled data that the algorithm tries to make sense of by extracting features and patterns on its own.


**D** The independent variables in this case are  labels- 

1. Pregnancies	
2. Glucose
3. BloodPressure
4. SkinThickness
5. Insulin
6. BMI
7. DiabetesPedigreeFunction
8. Age




This is so because these variable/labels embedded in the algorithm 
increase or decrease the probability/likhehood of the event
"*Outcome*".


**E** The dependent variable in this case is the label *Outcome* since it is the conclusive/final event or variable that predicts the likelihood of the occurence of diabetes.

**F** Here the categorical variables are - *

1.  outcome



This is so because a categorical variable is a variable that can take on one of a limited, and usually fixed, number of possible values, assigning each individual or other unit of observation to a particular group or nominal category on the basis of some qualitative property.



**G** The categorical variables follow *Label Encoding* where each label is mapped to a unique quantitative value.

**H**

In [ ]:
pd.cut(x=diabetes_df['Age'], bins=[21, 40, 60, 110]).value_counts()

(21, 40]     511
(40, 60]     167
(60, 110]     27
Name: Age, dtype: int64

Young Adults : 511 samples

1.   Young Adults : 511 samples
2.   Middle-ages Adults : 167 samples
3.   Old-aged Adults: 27



In [ ]:
diabetes_df.dtypes

Pregnancies                   int64
Glucose                       int64
BloodPressure                 int64
SkinThickness                 int64
Insulin                       int64
BMI                         float64
DiabetesPedigreeFunction    float64
Age                           int64
Outcome                       int64
dtype: object

# *2.2*

In [ ]:
diabetes_df.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

In [44]:
diabetes_df.replace(0, np.NaN, inplace=True)

In [45]:
diabetes_df.head(20)

Pregnancies  Glucose  BloodPressure  ...  DiabetesPedigreeFunction  Age  Outcome
0           6.0    148.0           72.0  ...                     0.627   50      1.0
1           1.0     85.0           66.0  ...                     0.351   31      NaN
2           8.0    183.0           64.0  ...                     0.672   32      1.0
3           1.0     89.0           66.0  ...                     0.167   21      NaN
4           NaN    137.0           40.0  ...                     2.288   33      1.0
5           5.0    116.0           74.0  ...                     0.201   30      NaN
6           3.0     78.0           50.0  ...                     0.248   26      1.0
7          10.0    115.0            NaN  ...                     0.134   29      NaN
8           2.0    197.0           70.0  ...                     0.158   53      1.0
9           8.0    125.0           96.0  ...                     0.232   54      1.0
10          4.0    110.0           92.0  ...                     0.191   30      NaN
11         10.0    168.0           74.0  ...                     0.537   34      1.0
12         10.0    139.0           80.0  ...                     1.441   57      NaN
13          1.0    189.0           60.0  ...                     0.398   59      1.0
14          5.0    166.0           72.0  ...                     0.587   51      1.0
15          7.0    100.0            NaN  ...                     0.484   32      1.0
16          NaN    118.0           84.0  ...                     0.551   31      1.0
17          7.0    107.0           74.0  ...                     0.254   31      1.0
18          1.0    103.0           30.0  ...                     0.183   33      NaN
19          1.0    115.0           70.0  ...                     0.529   32      1.0

[20 rows x 9 columns]

The dataset didn't have missing values in it. We can clearly see here the inconsistent values were a bunch of zeroes by itself. We replaced them with NaN.

In [46]:
diabetes_df.dtypes

Pregnancies                 float64
Glucose                     float64
BloodPressure               float64
SkinThickness               float64
Insulin                     float64
BMI                         float64
DiabetesPedigreeFunction    float64
Age                           int64
Outcome                     float64
dtype: object

We convert all the float back to int. Since that is not what we want.

In [47]:
diabetes_df.astype(int)

ValueError: ignored

# *2.3*

The Non Categorical columns in this dataset are -


1.   Pregnancies
2.   Glucose
3.   BloodPressure
4.   SkinThickness
5.   Insulin
6.   BMI
7.   DiabetesPedigreeFunction
8.   Age




In [37]:

heart_df_NC = heart_df[['age','creatinine_phosphokinase','ejection_fraction','platelets','serum_creatinine','serum_sodium','time']]
[ ]
avg = heart_df_NC.mean()
[ ]
stdv = heart_df_NC.std()
[ ]
standardized = (heart_df_NC - avg) / stdv
[ ]


NameError: ignored

In [48]:
diabetes_df = diabetes_df[['Pregnancies','Glucose','BloodPressure','SkinThickness','Insulin','BMI','DiabetesPedigreeFunction','Age']]

In [49]:
avg = diabetes_df.mean()

In [50]:
stdv = diabetes_df.std()


In [51]:
standardized = (diabetes_df - avg) / stdv

In [53]:
standardized.head(20)

Pregnancies   Glucose  ...  DiabetesPedigreeFunction       Age
0      0.467887  0.861722  ...                  0.468187  1.425067
1     -1.086216 -1.201441  ...                 -0.364823 -0.190548
2      1.089528  2.007924  ...                  0.604004 -0.105515
3     -1.086216 -1.070446  ...                 -0.920163 -1.040871
4           NaN  0.501487  ...                  5.481337 -0.020483
5      0.157066 -0.186234  ...                 -0.817546 -0.275580
6     -0.464575 -1.430681  ...                 -0.675693 -0.615709
7      1.711168 -0.218982  ...                 -1.019762 -0.360612
8     -0.775395  2.466404  ...                 -0.947326  1.680164
9      1.089528  0.108504  ...                 -0.723983  1.765196
10    -0.153754 -0.382725  ...                 -0.847727 -0.275580
11     1.711168  1.516694  ...                  0.196553  0.064549
12     1.711168  0.566985  ...                  2.924962  2.020293
13    -1.086216  2.204415  ...                 -0.222970  2.190358
14     0.157066  1.451197  ...                  0.347461  1.510099
15     0.778707 -0.710211  ...                  0.036591 -0.105515
16          NaN -0.120736  ...                  0.238807 -0.190548
17     0.778707 -0.480971  ...                 -0.657584 -0.190548
18    -1.086216 -0.611966  ...                 -0.871873 -0.020483
19    -1.086216 -0.218982  ...                  0.172408 -0.105515

[20 rows x 8 columns]

In [54]:
standardized.mean()

Pregnancies                -1.067977e-16
Glucose                     1.002547e-16
BloodPressure              -5.239480e-16
SkinThickness              -3.006426e-17
Insulin                    -6.903671e-18
BMI                         3.449978e-15
DiabetesPedigreeFunction    1.913039e-15
Age                         2.198762e-16
dtype: float64

In [55]:
standardized.var()

Pregnancies                 1.0
Glucose                     1.0
BloodPressure               1.0
SkinThickness               1.0
Insulin                     1.0
BMI                         1.0
DiabetesPedigreeFunction    1.0
Age                         1.0
dtype: float64

In [ ]:
diabetes_df = diabetes_df.rename(columns = {"DEATH_EVENT": "1"})